compiler/compile/OMRCompilation.cpp

```diff
@@ -2079,6 +2081,63 @@ void OMR::Compilation::verifyCFG(TR::ResolvedMethodSymbol *methodSymbol)
       }
    }
 
+
+/*
+ * \brief
+ *    Check to make sure the rdbars are anchored and anchor a rdbar if it's found unanchored
+ *
+ * \note
+ *    Ideally all optimizations should anchor a rdbar when it's created.
+ *    This API is for fixing the unanchored rdbars in case some optimizations forget to
+ *    anchor the node by mistake. Optimizations should still try to generate correct rdbar
+ *    trees instead of relying on this API.
+ */
+void OMR::Compilation::verifyAndFixRdbarAnchors()
+   {
+   TR::NodeChecklist anchoredRdbarNodes(self());
+   TR::NodeChecklist visitedNodes(self());
+   NodeTreeTopPairList unanchoredRdbarNodes(self()->trMemory()->currentStackRegion());
+   for (TR::TreeTop *tt = self()->getStartTree(); tt; tt = tt->getNextTreeTop())
```

Comment:
I wonder if the code would be cleaner if it used IL iterators from `infra/ILWalk.hpp` instead.

Reply:
Didn't know the API exists. Changed to use the node iterator to make the code more clear.